% Generated by Sphinx.
\def\sphinxdocclass{report}
\documentclass[letterpaper,10pt,english]{sphinxmanual}
\usepackage[utf8]{inputenc}
\DeclareUnicodeCharacter{00A0}{\nobreakspace}
\usepackage{cmap}
\usepackage[T1]{fontenc}
\usepackage{babel}
\usepackage{times}
\usepackage[Sonny]{fncychap}
\usepackage{longtable}
\usepackage{sphinx}
\usepackage{multirow}
\usepackage{eqparbox}


\addto\captionsenglish{\renewcommand{\figurename}{Fig. }}
\addto\captionsenglish{\renewcommand{\tablename}{Table }}
\SetupFloatingEnvironment{literal-block}{name=Listing }



\title{iCallSV Documentation}
\date{June 06, 2016}
\release{}
\author{Author}
\newcommand{\sphinxlogo}{}
\renewcommand{\releasename}{Release}
\setcounter{tocdepth}{3}
\makeindex

\makeatletter
\def\PYG@reset{\let\PYG@it=\relax \let\PYG@bf=\relax%
    \let\PYG@ul=\relax \let\PYG@tc=\relax%
    \let\PYG@bc=\relax \let\PYG@ff=\relax}
\def\PYG@tok#1{\csname PYG@tok@#1\endcsname}
\def\PYG@toks#1+{\ifx\relax#1\empty\else%
    \PYG@tok{#1}\expandafter\PYG@toks\fi}
\def\PYG@do#1{\PYG@bc{\PYG@tc{\PYG@ul{%
    \PYG@it{\PYG@bf{\PYG@ff{#1}}}}}}}
\def\PYG#1#2{\PYG@reset\PYG@toks#1+\relax+\PYG@do{#2}}

\expandafter\def\csname PYG@tok@gd\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.63,0.00,0.00}{##1}}}
\expandafter\def\csname PYG@tok@gu\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.50,0.00,0.50}{##1}}}
\expandafter\def\csname PYG@tok@gt\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.27,0.87}{##1}}}
\expandafter\def\csname PYG@tok@gs\endcsname{\let\PYG@bf=\textbf}
\expandafter\def\csname PYG@tok@gr\endcsname{\def\PYG@tc##1{\textcolor[rgb]{1.00,0.00,0.00}{##1}}}
\expandafter\def\csname PYG@tok@cm\endcsname{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.25,0.50,0.56}{##1}}}
\expandafter\def\csname PYG@tok@vg\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.73,0.38,0.84}{##1}}}
\expandafter\def\csname PYG@tok@vi\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.73,0.38,0.84}{##1}}}
\expandafter\def\csname PYG@tok@mh\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\expandafter\def\csname PYG@tok@cs\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.50,0.56}{##1}}\def\PYG@bc##1{\setlength{\fboxsep}{0pt}\colorbox[rgb]{1.00,0.94,0.94}{\strut ##1}}}
\expandafter\def\csname PYG@tok@ge\endcsname{\let\PYG@it=\textit}
\expandafter\def\csname PYG@tok@vc\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.73,0.38,0.84}{##1}}}
\expandafter\def\csname PYG@tok@il\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\expandafter\def\csname PYG@tok@go\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.20,0.20,0.20}{##1}}}
\expandafter\def\csname PYG@tok@cp\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@gi\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.63,0.00}{##1}}}
\expandafter\def\csname PYG@tok@gh\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.00,0.50}{##1}}}
\expandafter\def\csname PYG@tok@ni\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.84,0.33,0.22}{##1}}}
\expandafter\def\csname PYG@tok@nl\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.13,0.44}{##1}}}
\expandafter\def\csname PYG@tok@nn\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.05,0.52,0.71}{##1}}}
\expandafter\def\csname PYG@tok@no\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.38,0.68,0.84}{##1}}}
\expandafter\def\csname PYG@tok@na\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\expandafter\def\csname PYG@tok@nb\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@nc\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.05,0.52,0.71}{##1}}}
\expandafter\def\csname PYG@tok@nd\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.33,0.33,0.33}{##1}}}
\expandafter\def\csname PYG@tok@ne\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@nf\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.02,0.16,0.49}{##1}}}
\expandafter\def\csname PYG@tok@si\endcsname{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.44,0.63,0.82}{##1}}}
\expandafter\def\csname PYG@tok@s2\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\expandafter\def\csname PYG@tok@nt\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.02,0.16,0.45}{##1}}}
\expandafter\def\csname PYG@tok@nv\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.73,0.38,0.84}{##1}}}
\expandafter\def\csname PYG@tok@s1\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\expandafter\def\csname PYG@tok@ch\endcsname{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.25,0.50,0.56}{##1}}}
\expandafter\def\csname PYG@tok@m\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\expandafter\def\csname PYG@tok@gp\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.78,0.36,0.04}{##1}}}
\expandafter\def\csname PYG@tok@sh\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\expandafter\def\csname PYG@tok@ow\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@sx\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.78,0.36,0.04}{##1}}}
\expandafter\def\csname PYG@tok@bp\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@c1\endcsname{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.25,0.50,0.56}{##1}}}
\expandafter\def\csname PYG@tok@o\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.40,0.40,0.40}{##1}}}
\expandafter\def\csname PYG@tok@kc\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@c\endcsname{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.25,0.50,0.56}{##1}}}
\expandafter\def\csname PYG@tok@mf\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\expandafter\def\csname PYG@tok@err\endcsname{\def\PYG@bc##1{\setlength{\fboxsep}{0pt}\fcolorbox[rgb]{1.00,0.00,0.00}{1,1,1}{\strut ##1}}}
\expandafter\def\csname PYG@tok@mb\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\expandafter\def\csname PYG@tok@ss\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.32,0.47,0.09}{##1}}}
\expandafter\def\csname PYG@tok@sr\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.14,0.33,0.53}{##1}}}
\expandafter\def\csname PYG@tok@mo\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\expandafter\def\csname PYG@tok@kd\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@mi\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.13,0.50,0.31}{##1}}}
\expandafter\def\csname PYG@tok@kn\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@cpf\endcsname{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.25,0.50,0.56}{##1}}}
\expandafter\def\csname PYG@tok@kr\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@s\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\expandafter\def\csname PYG@tok@kp\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@w\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.73,0.73,0.73}{##1}}}
\expandafter\def\csname PYG@tok@kt\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.56,0.13,0.00}{##1}}}
\expandafter\def\csname PYG@tok@sc\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\expandafter\def\csname PYG@tok@sb\endcsname{\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\expandafter\def\csname PYG@tok@k\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.00,0.44,0.13}{##1}}}
\expandafter\def\csname PYG@tok@se\endcsname{\let\PYG@bf=\textbf\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}
\expandafter\def\csname PYG@tok@sd\endcsname{\let\PYG@it=\textit\def\PYG@tc##1{\textcolor[rgb]{0.25,0.44,0.63}{##1}}}

\def\PYGZbs{\char`\\}
\def\PYGZus{\char`\_}
\def\PYGZob{\char`\{}
\def\PYGZcb{\char`\}}
\def\PYGZca{\char`\^}
\def\PYGZam{\char`\&}
\def\PYGZlt{\char`\<}
\def\PYGZgt{\char`\>}
\def\PYGZsh{\char`\#}
\def\PYGZpc{\char`\%}
\def\PYGZdl{\char`\$}
\def\PYGZhy{\char`\-}
\def\PYGZsq{\char`\'}
\def\PYGZdq{\char`\"}
\def\PYGZti{\char`\~}
% for compatibility with earlier versions
\def\PYGZat{@}
\def\PYGZlb{[}
\def\PYGZrb{]}
\makeatother

\renewcommand\PYGZsq{\textquotesingle}

\begin{document}

\maketitle
\tableofcontents
\phantomsection\label{index::doc}


Contents:


\chapter{iCallSV package}
\label{iCallSV:welcome-to-icallsv-s-documentation}\label{iCallSV::doc}\label{iCallSV:icallsv-package}

\section{Submodules}
\label{iCallSV:submodules}

\section{iCallSV.FilterDellyCalls module}
\label{iCallSV:icallsv-filterdellycalls-module}\label{iCallSV:module-iCallSV.FilterDellyCalls}\index{iCallSV.FilterDellyCalls (module)}

\subsection{FilterDellyCalls}
\label{iCallSV:filterdellycalls}\begin{quote}\begin{description}
\item[{Description}] \leavevmode
This module will filter calls made by Delly which are in a VCF format

\end{description}\end{quote}
\index{GetCaseFlag() (in module iCallSV.FilterDellyCalls)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.FilterDellyCalls.GetCaseFlag}\pysiglinewithargsret{\code{iCallSV.FilterDellyCalls.}\bfcode{GetCaseFlag}}{\emph{caseDR}, \emph{caseDV}, \emph{preciseFlag}, \emph{caseRR}, \emph{caseRV}, \emph{caseAltFreq}, \emph{caseTotalCount}}{}
This will \code{check if the case sample passes or not}
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{caseDR}} (\emph{\texttt{int}}) -- int representing number of reference reads for case reported by delly

\item {} 
\textbf{\texttt{caseDV}} (\emph{\texttt{int}}) -- int representing number of variant reads for case reported by delly

\item {} 
\textbf{\texttt{preciseFlag}} (\emph{\texttt{str}}) -- str representing if an event is precise or imprecise

\item {} 
\textbf{\texttt{caseRR}} (\emph{\texttt{int}}) -- int representing number of split reference reads for case reported by delly

\item {} 
\textbf{\texttt{caseRV}} (\emph{\texttt{int}}) -- int representing number of split variant reads for case reported by delly

\item {} 
\textbf{\texttt{caseAltFreq}} (\emph{\texttt{float}}) -- float representing altratio threshold for case

\item {} 
\textbf{\texttt{caseTotalCount}} (\emph{\texttt{int}}) -- int repeseting readcount threshold for case

\end{itemize}

\item[{Returns}] \leavevmode
A boolean tag indicating True or False

\item[{Return type}] \leavevmode
bool

\end{description}\end{quote}

\end{fulllineitems}

\index{GetControlFlag() (in module iCallSV.FilterDellyCalls)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.FilterDellyCalls.GetControlFlag}\pysiglinewithargsret{\code{iCallSV.FilterDellyCalls.}\bfcode{GetControlFlag}}{\emph{controlDR}, \emph{controlDV}, \emph{preciseFlag}, \emph{controlRR}, \emph{controlRV}, \emph{controlAltFreq}}{}
This will \code{check if the control sample passes or not}
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{controlDR}} (\emph{\texttt{int}}) -- int representing number of reference reads for control reported by delly

\item {} 
\textbf{\texttt{controlDV}} (\emph{\texttt{int}}) -- int representing number of variant reads for control reported by delly

\item {} 
\textbf{\texttt{preciseFlag}} (\emph{\texttt{str}}) -- str representing if an event is precise or imprecise

\item {} 
\textbf{\texttt{controlRR}} (\emph{\texttt{int}}) -- int representing number of split reference reads for control reported by delly

\item {} 
\textbf{\texttt{controlRV}} (\emph{\texttt{int}}) -- int representing number of split variant reads for control reported by delly

\item {} 
\textbf{\texttt{controlAltFreq}} (\emph{\texttt{float}}) -- float representing altratio threshold for control

\end{itemize}

\item[{Returns}] \leavevmode
A boolean tag indicating True or False

\item[{Return type}] \leavevmode
bool

\end{description}\end{quote}

\end{fulllineitems}

\index{GetFilteredRecords() (in module iCallSV.FilterDellyCalls)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.FilterDellyCalls.GetFilteredRecords}\pysiglinewithargsret{\code{iCallSV.FilterDellyCalls.}\bfcode{GetFilteredRecords}}{\emph{dellyVarialbles}, \emph{thresholdVariables}, \emph{hotspotDict}, \emph{blacklist}}{}
This will \code{Filter one record at a time}
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{dellyVariables}} (\emph{\texttt{str}}) -- str having all delly variables separated by '',''

\item {} 
\textbf{\texttt{thresholdVariables}} (\emph{\texttt{str}}) -- str having all delly threshold variables separated by '',''

\item {} 
\textbf{\texttt{hotspotDict}} (\emph{\texttt{dict}}) -- A dict containing hotspot regions

\item {} 
\textbf{\texttt{blacklist}} (\emph{\texttt{list}}) -- A list containing blacklist regions

\end{itemize}

\item[{Returns}] \leavevmode
A boolean tag indicating True or False

\item[{Return type}] \leavevmode
bool

\end{description}\end{quote}

\end{fulllineitems}

\index{run() (in module iCallSV.FilterDellyCalls)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.FilterDellyCalls.run}\pysiglinewithargsret{\code{iCallSV.FilterDellyCalls.}\bfcode{run}}{\emph{inputVcf}, \emph{outputDir}, \emph{controlId}, \emph{caseID}, \emph{hotspotFile}, \emph{blacklistFile}, \emph{svlength}, \emph{mapq}, \emph{mapqHotspot}, \emph{caseAltFreqHotspot}, \emph{caseTotalCountHotspot}, \emph{controlAltFreqHotspot}, \emph{caseAltFreq}, \emph{caseTotalCount}, \emph{controlAltFreq}, \emph{peSupport}, \emph{srSupport}, \emph{peSupportHotspot}, \emph{srSupportHotspot}, \emph{peSupportCase}, \emph{srSupportCase}, \emph{peSupportHotspotCase}, \emph{srSupportHotspotCase}, \emph{peSupportControl}, \emph{srSupportControl}, \emph{peSupportHotspotControl}, \emph{srSupportHotspotControl}, \emph{verbose}}{}
{\color{red}\bfseries{}{}`{}`}main:{\color{red}\bfseries{}{}`{}`}Filter calls made by Delly which are in a VCF format
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{inputVcf}} (\emph{\texttt{str}}) -- Input VCF file name with path

\item {} 
\textbf{\texttt{outputDir}} (\emph{\texttt{str}}) -- Output directory

\item {} 
\textbf{\texttt{controlId}} (\emph{\texttt{str}}) -- Control Sample ID (Should be part of Sample Name in VCF)

\item {} 
\textbf{\texttt{caseID}} (\emph{\texttt{str}}) -- Case Sample ID (Should be part of Sample Name in VCF)

\item {} 
\textbf{\texttt{hospotFile}} (\emph{\texttt{str}}) -- List of Genes that have Hotspot Structural Variants (Tab-delimited Format without header:chr    start    end    geneName).

\item {} 
\textbf{\texttt{blacklistFile}} (\emph{\texttt{str}}) -- List of Genes that have blacklist of Structural Variants (Tab-delimited Format without header:chr    start1    chr2     start2; where chr1==chr2, end==start2).

\item {} 
\textbf{\texttt{caseAltFreq}} (\emph{\texttt{float}}) -- Alternate Allele Frequency threshold for case

\item {} 
\textbf{\texttt{caseTotalCount}} (\emph{\texttt{int}}) -- Total ReadCount threshold for case

\item {} 
\textbf{\texttt{ccontrolAltFreq}} (\emph{\texttt{flaot}}) -- Alternate Allele Frequency threshold for control

\item {} 
\textbf{\texttt{peSupport}} (\emph{\texttt{int}}) -- overall pair-end read support threshold for the event

\item {} 
\textbf{\texttt{srSupport}} (\emph{\texttt{int}}) -- overall split-reads support threshold for the event

\item {} 
\textbf{\texttt{peSupportHotspot}} (\emph{\texttt{int}}) -- overall pair-end read support threshold for the event in hot-spot region

\item {} 
\textbf{\texttt{srSupportHotspot}} (\emph{\texttt{int}}) -- overall split-reads support threshold for the event in hot-spot region

\item {} 
\textbf{\texttt{peSupportCase}} (\emph{\texttt{int}}) -- pair-end read support threshold for the event in the Case sample

\item {} 
\textbf{\texttt{srSupportCase}} (\emph{\texttt{int}}) -- split-reads support threshold for the event in the Case sample

\item {} 
\textbf{\texttt{peSupportHotspotCase}} (\emph{\texttt{int}}) -- pair-end read support threshold for the event in hot-spot region for the Case sample

\item {} 
\textbf{\texttt{srSupportHotspotCase}} (\emph{\texttt{int}}) -- split-reads support threshold for the event in hot-spot region for the Case sample

\item {} 
\textbf{\texttt{peSupportControl}} (\emph{\texttt{int}}) -- pair-end read support threshold for the event in the Control sample

\item {} 
\textbf{\texttt{srSupportControl}} (\emph{\texttt{int}}) -- split-reads support threshold for the event in the Control sample

\item {} 
\textbf{\texttt{peSupportHotspotControl}} (\emph{\texttt{int}}) -- pair-end read support threshold for the event in hot-spot region for the Control sample

\item {} 
\textbf{\texttt{srSupportHotspotControl}} (\emph{\texttt{int}}) -- split-reads support threshold for the event in hot-spot region for the Control sample

\item {} 
\textbf{\texttt{svlength}} (\emph{\texttt{int}}) -- length of the structural variants

\item {} 
\textbf{\texttt{mapq}} (\emph{\texttt{int}}) -- overall mapping quality

\item {} 
\textbf{\texttt{mapqHotspot}} (\emph{\texttt{int}}) -- mapping quality for hot-spots

\end{itemize}

\item[{Returns}] \leavevmode
A str name of filtered vcf file

\item[{Return type}] \leavevmode
str

\end{description}\end{quote}

\end{fulllineitems}



\section{iCallSV.Run\_Delly module}
\label{iCallSV:module-iCallSV.Run_Delly}\label{iCallSV:icallsv-run-delly-module}\index{iCallSV.Run\_Delly (module)}

\subsection{Run\_Delly}
\label{iCallSV:run-delly}\begin{quote}\begin{description}
\item[{Description}] \leavevmode
Runs the delly program on case and control bam file to give its results

\end{description}\end{quote}
\index{run() (in module iCallSV.Run\_Delly)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.Run_Delly.run}\pysiglinewithargsret{\code{iCallSV.Run\_Delly.}\bfcode{run}}{\emph{delly}, \emph{version}, \emph{bcftools}, \emph{analysisType}, \emph{reference}, \emph{controlBam}, \emph{caseBam}, \emph{caseId}, \emph{mapq}, \emph{excludeRegions}, \emph{outputdir}, \emph{verbose}, \emph{debug}}{}
This will Runs the delly program on case and control bam file to give its
results.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{delly}} (\emph{\texttt{str}}) -- Path to delly executables (0.7.3 or above)

\item {} 
\textbf{\texttt{bcftools}} (\emph{\texttt{str}}) -- Path to bcftools executables (1.3.1 or above)

\item {} 
\textbf{\texttt{type}} (\emph{\texttt{str}}) -- What ot run in delly, DEL:Deletion, DUP: Duplication,TRA:Translocation, INV:Inversion

\item {} 
\textbf{\texttt{reference}} (\emph{\texttt{str}}) -- Reference Genome that was used to align the reads.

\item {} 
\textbf{\texttt{controlBam}} (\emph{\texttt{str}}) -- Path to control/normal bam file

\item {} 
\textbf{\texttt{caseBam}} (\emph{\texttt{str}}) -- Path to case/tumor bam file

\item {} 
\textbf{\texttt{controlID}} (\emph{\texttt{str}}) -- Id of the control/normal sample

\item {} 
\textbf{\texttt{caseID}} (\emph{\texttt{str}}) -- Id of the case/tumor sample

\item {} 
\textbf{\texttt{mapq}} (\emph{\texttt{int}}) -- mapping quality cutoff for delly

\item {} 
\textbf{\texttt{excludeRegions}} (\emph{\texttt{str}}) -- Regions to be excluded for calling structural variation.

\item {} 
\textbf{\texttt{outputdir}} (\emph{\texttt{str}}) -- directory for the output of delly

\item {} 
\textbf{\texttt{debug}} (\emph{\texttt{bool}}) -- If you just wish to test what we will do

\end{itemize}

\item[{Returns}] \leavevmode
str of the output vcf

\item[{Return type}] \leavevmode
str

\end{description}\end{quote}

\end{fulllineitems}



\section{iCallSV.Run\_iAnnotateSV module}
\label{iCallSV:icallsv-run-iannotatesv-module}\label{iCallSV:module-iCallSV.Run_iAnnotateSV}\index{iCallSV.Run\_iAnnotateSV (module)}

\subsection{Run\_iAnnotate}
\label{iCallSV:run-iannotate}\begin{quote}\begin{description}
\item[{Description}] \leavevmode
This module will run iAnnotateSV package

\end{description}\end{quote}
\index{run() (in module iCallSV.Run\_iAnnotateSV)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.Run_iAnnotateSV.run}\pysiglinewithargsret{\code{iCallSV.Run\_iAnnotateSV.}\bfcode{run}}{\emph{python}, \emph{iAnnotateSV}, \emph{build}, \emph{distance}, \emph{canonicalTranscriptFile}, \emph{uniprotFile}, \emph{cosmicFile}, \emph{repeatregionFile}, \emph{dgvFile}, \emph{inputTabFile}, \emph{outputPrefix}, \emph{outputDir}}{}
This module will run iAnnotateSV package.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{python}} (\emph{\texttt{str}}) -- Location for the python executable.

\item {} 
\textbf{\texttt{iAnnotateSV}} (\emph{\texttt{str}}) -- Location of the wrapper iAnnotateSV package (iAnnotateSV.py)

\item {} 
\textbf{\texttt{build}} (\emph{\texttt{str}}) -- Which human reference file to be used, hg18,hg19 or hg38

\item {} 
\textbf{\texttt{inputTabFile}} (\emph{\texttt{str}}) -- Tab-Delimited Input FIle compatible with iAnnotateSV package.

\item {} 
\textbf{\texttt{outputPrefix}} (\emph{\texttt{str}}) -- Prefix of the output files/DIR with Annotations and images

\item {} 
\textbf{\texttt{outputDir}} (\emph{\texttt{str}}) -- Name of the output directory where the outputPrefix will be written

\item {} 
\textbf{\texttt{uniprotFile}} (\emph{\texttt{str}}) -- Location for ucsc uniprot file

\item {} 
\textbf{\texttt{cosmicFile}} (\emph{\texttt{str}}) -- Location for cosmic census file

\item {} 
\textbf{\texttt{repeatregionFile}} (\emph{\texttt{str}}) -- Location for repeat region file

\item {} 
\textbf{\texttt{dgvFile}} (\emph{\texttt{str}}) -- Location for database of Genomic Variants file

\end{itemize}

\item[{Returns}] \leavevmode
str of the output file

\item[{Return type}] \leavevmode
str

\end{description}\end{quote}

\end{fulllineitems}



\section{iCallSV.Run\_samblaster module}
\label{iCallSV:icallsv-run-samblaster-module}\label{iCallSV:module-iCallSV.Run_samblaster}\index{iCallSV.Run\_samblaster (module)}

\subsection{Run\_samblaster}
\label{iCallSV:run-samblaster}\begin{quote}\begin{description}
\item[{Description}] \leavevmode
This module will run samblaster for extracting discordant and spit reads in sam format

\end{description}\end{quote}
\index{run() (in module iCallSV.Run\_samblaster)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.Run_samblaster.run}\pysiglinewithargsret{\code{iCallSV.Run\_samblaster.}\bfcode{run}}{\emph{samtools}, \emph{samblaster}, \emph{bamFile}, \emph{discordantFileName}, \emph{splitFileName}, \emph{outputDir}}{}
\end{fulllineitems}



\section{iCallSV.Run\_targetSeqView module}
\label{iCallSV:module-iCallSV.Run_targetSeqView}\label{iCallSV:icallsv-run-targetseqview-module}\index{iCallSV.Run\_targetSeqView (module)}

\subsection{Run\_targetSeqView}
\label{iCallSV:run-targetseqview}\begin{quote}\begin{description}
\item[{Description}] \leavevmode
This module will run targetSeqView

\end{description}\end{quote}
\index{run() (in module iCallSV.Run\_targetSeqView)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.Run_targetSeqView.run}\pysiglinewithargsret{\code{iCallSV.Run\_targetSeqView.}\bfcode{run}}{\emph{RLocation}, \emph{targetSeqView}, \emph{nodes}, \emph{bamFilePath}, \emph{svFile}, \emph{build}, \emph{readLength}, \emph{outputDir}, \emph{outsvFileName}}{}
This module will run targetSeqView.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{RLocation}} (\emph{\texttt{str}}) -- Location of the R executable (\textgreater{}3.1.2).

\item {} 
\textbf{\texttt{targetSeqView}} (\emph{\texttt{str}}) -- Location of R script that will run tragetSeqView

\item {} 
\textbf{\texttt{nodes}} (\emph{\texttt{int}}) -- Number of parallel nodes for running targetSeqView

\item {} 
\textbf{\texttt{bamFile}} (\emph{\texttt{str}}) -- Location of the bamFile which has the  structural variant events.

\item {} 
\textbf{\texttt{svFile}} (\emph{\texttt{str}}) -- targetSeqView compatible input structural variant file.

\item {} 
\textbf{\texttt{build}} (\emph{\texttt{str}}) -- Which human reference file to be used, hg18,hg19 or hg38

\item {} 
\textbf{\texttt{readLength}} (\emph{\texttt{int}}) -- Sequencing Read Length (101)

\item {} 
\textbf{\texttt{outputDir}} (\emph{\texttt{str}}) -- Directory for output files

\item {} 
\textbf{\texttt{outsvFile}} (\emph{\texttt{str}}) -- Name of the output structural variant file that has added confidence score to it.

\end{itemize}

\item[{Returns}] \leavevmode
str of the output file

\item[{Return type}] \leavevmode
str

\end{description}\end{quote}

\end{fulllineitems}



\section{iCallSV.checkBlackList module}
\label{iCallSV:icallsv-checkblacklist-module}\label{iCallSV:module-iCallSV.checkBlackList}\index{iCallSV.checkBlackList (module)}

\subsection{checkBlackList}
\label{iCallSV:checkblacklist}\begin{quote}\begin{description}
\item[{Description}] \leavevmode
This module will read the Black List file and tell if and event is blacklisted or not

\end{description}\end{quote}
\index{CheckIfItIsBlacklisted() (in module iCallSV.checkBlackList)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.checkBlackList.CheckIfItIsBlacklisted}\pysiglinewithargsret{\code{iCallSV.checkBlackList.}\bfcode{CheckIfItIsBlacklisted}}{\emph{chr1}, \emph{start1}, \emph{chr2}, \emph{start2}, \emph{blacklist}, \emph{extention}}{}
Check if coordinate are present in the \code{blacklist region file}
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{chr1}} (\emph{\texttt{str}}) -- str of the breakpoint in first chromosome

\item {} 
\textbf{\texttt{start1}} (\emph{\texttt{int}}) -- int of the start location of the breakpoint in first chromosome

\item {} 
\textbf{\texttt{chr2}} (\emph{\texttt{str}}) -- str of the breakpoint in second chromosome

\item {} 
\textbf{\texttt{start2}} (\emph{\texttt{int}}) -- int of the start location of the breakpoint in second chromosome

\item {} 
\textbf{\texttt{blacklist}} (\emph{\texttt{list}}) -- A list containing black listed regions

\item {} 
\textbf{\texttt{extension}} (\emph{\texttt{int}}) -- an value for search in positive and negative direction of the start1 and start2 location

\end{itemize}

\item[{Returns}] \leavevmode
A boolean tag indicating True or False

\item[{Return type}] \leavevmode
bool

\end{description}\end{quote}

\end{fulllineitems}

\index{ReadBlackListFile() (in module iCallSV.checkBlackList)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.checkBlackList.ReadBlackListFile}\pysiglinewithargsret{\code{iCallSV.checkBlackList.}\bfcode{ReadBlackListFile}}{\emph{BlackListFile}}{}
Read the \code{blacklist region file}
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\textbf{\texttt{BlackListFile}} (\emph{\texttt{str}}) -- str of file to be read.

\item[{Returns}] \leavevmode
A list containing black listed regions.

\item[{Return type}] \leavevmode
list.

\end{description}\end{quote}

\end{fulllineitems}



\section{iCallSV.checkHotSpotList module}
\label{iCallSV:icallsv-checkhotspotlist-module}\label{iCallSV:module-iCallSV.checkHotSpotList}\index{iCallSV.checkHotSpotList (module)}

\subsection{checkHotSpotList}
\label{iCallSV:checkhotspotlist}\begin{quote}\begin{description}
\item[{Description}] \leavevmode
This module will read the hotspot file and tell if it is a hotspot or not

\end{description}\end{quote}
\index{CheckIfItIsHotspot() (in module iCallSV.checkHotSpotList)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.checkHotSpotList.CheckIfItIsHotspot}\pysiglinewithargsret{\code{iCallSV.checkHotSpotList.}\bfcode{CheckIfItIsHotspot}}{\emph{chr1}, \emph{start1}, \emph{chr2}, \emph{start2}, \emph{hotspotDict}}{}
Check if coordinate are present in the \code{hotspot region file}
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{chr1}} (\emph{\texttt{str}}) -- str of the breakpoint in first chromosome

\item {} 
\textbf{\texttt{start1}} (\emph{\texttt{int}}) -- int of the start location of the breakpoint in first chromosome

\item {} 
\textbf{\texttt{chr2}} (\emph{\texttt{str}}) -- str of the breakpoint in second chromosome

\item {} 
\textbf{\texttt{start2}} (\emph{\texttt{int}}) -- int of the start location of the breakpoint in second chromosome

\item {} 
\textbf{\texttt{hotspotDict}} (\emph{\texttt{dict}}) -- A dict containing hotspot regions

\end{itemize}

\item[{Returns}] \leavevmode
A boolean tag indicating True or False

\item[{Return type}] \leavevmode
bool

\end{description}\end{quote}

\end{fulllineitems}

\index{ReadHotSpotFile() (in module iCallSV.checkHotSpotList)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.checkHotSpotList.ReadHotSpotFile}\pysiglinewithargsret{\code{iCallSV.checkHotSpotList.}\bfcode{ReadHotSpotFile}}{\emph{HotSpotFile}}{}
Read the \code{HotSpot region file}
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\textbf{\texttt{HotSpotFile}} (\emph{\texttt{str}}) -- str of file to be read.

\item[{Returns}] \leavevmode
A dict containing hotspot regions

\item[{Return type}] \leavevmode
dict

\end{description}\end{quote}

\end{fulllineitems}



\section{iCallSV.checkparameters module}
\label{iCallSV:icallsv-checkparameters-module}\label{iCallSV:module-iCallSV.checkparameters}\index{iCallSV.checkparameters (module)}

\subsection{checkparameters}
\label{iCallSV:checkparameters}\begin{quote}\begin{description}
\item[{Description}] \leavevmode
This modules checks the parameters for various type of inputs.

\end{description}\end{quote}
\index{checkDellyAnalysisType() (in module iCallSV.checkparameters)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.checkparameters.checkDellyAnalysisType}\pysiglinewithargsret{\code{iCallSV.checkparameters.}\bfcode{checkDellyAnalysisType}}{\emph{varaibleToCheck}}{}
Check \emph{if the variable for Delly analysis exists or not{}`}
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\textbf{\texttt{variableToCheck}} (\emph{\texttt{str}}) -- check if str is DEL\textbar{}DUP\textbar{}INV\textbar{}TRA

\item[{Returns}] \leavevmode
None

\item[{Return type}] \leavevmode
None

\end{description}\end{quote}

\end{fulllineitems}

\index{checkDir() (in module iCallSV.checkparameters)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.checkparameters.checkDir}\pysiglinewithargsret{\code{iCallSV.checkparameters.}\bfcode{checkDir}}{\emph{folderToCheck}}{}
Check \emph{if the folder exists or not{}`}

\code{str}.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\textbf{\texttt{folderToCheck}} (\emph{\texttt{str}}) -- Name of the folder to be checked.

\item[{Returns}] \leavevmode
None

\item[{Return type}] \leavevmode
None

\end{description}\end{quote}

\end{fulllineitems}

\index{checkEmpty() (in module iCallSV.checkparameters)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.checkparameters.checkEmpty}\pysiglinewithargsret{\code{iCallSV.checkparameters.}\bfcode{checkEmpty}}{\emph{variableToCheck}, \emph{variableName}}{}
Check \emph{if the variable is None or not{}`}
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{variableToCheck}} (\emph{\texttt{str}}) -- check if str is None or not

\item {} 
\textbf{\texttt{variableName}} (\emph{\texttt{str}}) -- Name of the None object to be verified

\end{itemize}

\item[{Returns}] \leavevmode
None

\item[{Return type}] \leavevmode
None

\end{description}\end{quote}

\end{fulllineitems}

\index{checkFile() (in module iCallSV.checkparameters)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.checkparameters.checkFile}\pysiglinewithargsret{\code{iCallSV.checkparameters.}\bfcode{checkFile}}{\emph{fileToCheck}}{}
Check \emph{if the file exists or not{}`}
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\textbf{\texttt{fileToCheck}} (\emph{\texttt{str}}) -- Name of the file to be checked.

\item[{Returns}] \leavevmode
None

\item[{Return type}] \leavevmode
None

\end{description}\end{quote}

\end{fulllineitems}

\index{checkInt() (in module iCallSV.checkparameters)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.checkparameters.checkInt}\pysiglinewithargsret{\code{iCallSV.checkparameters.}\bfcode{checkInt}}{\emph{variableToCheck}, \emph{variableName}}{}
Check \emph{if the variable is int or not{}`}
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{variableToCheck}} (\emph{\texttt{int}}) -- Check if it is int or not

\item {} 
\textbf{\texttt{variableName}} (\emph{\texttt{str}}) -- Name of the int object to be verified

\end{itemize}

\item[{Returns}] \leavevmode
None

\item[{Return type}] \leavevmode
None

\end{description}\end{quote}

\end{fulllineitems}



\section{iCallSV.combineVCF module}
\label{iCallSV:module-iCallSV.combineVCF}\label{iCallSV:icallsv-combinevcf-module}\index{iCallSV.combineVCF (module)}

\subsection{combineVCF}
\label{iCallSV:combinevcf}\begin{quote}\begin{description}
\item[{Description}] \leavevmode
This module will combine multiple vcf file with same headers

\end{description}\end{quote}
\index{run() (in module iCallSV.combineVCF)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.combineVCF.run}\pysiglinewithargsret{\code{iCallSV.combineVCF.}\bfcode{run}}{\emph{vcfFiles}, \emph{combinedVCF}, \emph{verbose}}{}
This will \code{combine multiple vcf file with same headers}
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{vcfFiles}} (\emph{\texttt{list}}) -- a list of .vcf files to be combined

\item {} 
\textbf{\texttt{combinedVCF}} (\emph{\texttt{str}}) -- str for the output of combined vcf files

\item {} 
\textbf{\texttt{verbose}} (\emph{\texttt{bool}}) -- a boolean

\end{itemize}

\item[{Returns}] \leavevmode
A str name of combined vcf file

\item[{Return type}] \leavevmode
str

\end{description}\end{quote}

\end{fulllineitems}



\section{iCallSV.dellyVcf2Tab module}
\label{iCallSV:module-iCallSV.dellyVcf2Tab}\label{iCallSV:icallsv-dellyvcf2tab-module}\index{iCallSV.dellyVcf2Tab (module)}

\subsection{dellyVcf2Tab}
\label{iCallSV:dellyvcf2tab}\begin{quote}\begin{description}
\item[{Description}] \leavevmode
This module converts the Delly Vcf file having tumor normal, to tab-delimited format for input to iAnnotateSV

\end{description}\end{quote}
\index{vcf2tab() (in module iCallSV.dellyVcf2Tab)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.dellyVcf2Tab.vcf2tab}\pysiglinewithargsret{\code{iCallSV.dellyVcf2Tab.}\bfcode{vcf2tab}}{\emph{vcfFile}, \emph{outputDir}, \emph{verbose}}{}
This \code{converts} the Delly Vcf file having tumor normal, to tab-delimited format for input to iAnnotateSV
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{vcfFile}} (\emph{\texttt{str}}) -- str of vcf file to be converted

\item {} 
\textbf{\texttt{outputDir}} (\emph{\texttt{str}}) -- str for the output directory

\item {} 
\textbf{\texttt{verbose}} (\emph{\texttt{bool}}) -- a boolean

\end{itemize}

\item[{Returns}] \leavevmode
A str name of tab-delimited file

\item[{Return type}] \leavevmode
str

\end{description}\end{quote}

\end{fulllineitems}



\section{iCallSV.dellyVcf2targetSeqView module}
\label{iCallSV:module-iCallSV.dellyVcf2targetSeqView}\label{iCallSV:icallsv-dellyvcf2targetseqview-module}\index{iCallSV.dellyVcf2targetSeqView (module)}

\subsection{dellyVcf2targetSeqView}
\label{iCallSV:dellyvcf2targetseqview}\begin{quote}\begin{description}
\item[{Description}] \leavevmode
Convert VCF to targetSeqView

\end{description}\end{quote}
\index{Convert2targetSeqView() (in module iCallSV.dellyVcf2targetSeqView)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.dellyVcf2targetSeqView.Convert2targetSeqView}\pysiglinewithargsret{\code{iCallSV.dellyVcf2targetSeqView.}\bfcode{Convert2targetSeqView}}{\emph{sampleName}, \emph{sampleBamName}, \emph{sampleSplitBamName}, \emph{vcfFile}, \emph{outputDir}, \emph{outputFileName}}{}
This \code{converts} the Delly Vcf file having tumor normal, to tab-delimited format for input to targetSeqView
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{sampleName}} (\emph{\texttt{str}}) -- str for the name of the sample being analyzed

\item {} 
\textbf{\texttt{sampleBamName}} (\emph{\texttt{str}}) -- str for the pair-end reads bam file

\item {} 
\textbf{\texttt{sampleSplitBamName}} (\emph{\texttt{str}}) -- str for the split reads bam file

\item {} 
\textbf{\texttt{vcfFile}} (\emph{\texttt{str}}) -- str of vcf file to be converted

\item {} 
\textbf{\texttt{outputDir}} (\emph{\texttt{str}}) -- str for the output directory

\item {} 
\textbf{\texttt{outputFileName}} (\emph{\texttt{str}}) -- str for the output File

\end{itemize}

\item[{Returns}] \leavevmode
A str name of tab-delimited file

\item[{Return type}] \leavevmode
str

\end{description}\end{quote}

\end{fulllineitems}



\section{iCallSV.filterAnnotatedSV module}
\label{iCallSV:icallsv-filterannotatedsv-module}\label{iCallSV:module-iCallSV.filterAnnotatedSV}\index{iCallSV.filterAnnotatedSV (module)}

\subsection{filterAnnotatedSV}
\label{iCallSV:filterannotatedsv}\begin{quote}\begin{description}
\item[{Description}] \leavevmode
This module will filter calls from the merged file

\end{description}\end{quote}
\index{checkBlackListGene() (in module iCallSV.filterAnnotatedSV)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.filterAnnotatedSV.checkBlackListGene}\pysiglinewithargsret{\code{iCallSV.filterAnnotatedSV.}\bfcode{checkBlackListGene}}{\emph{gene1}, \emph{gene2}, \emph{blacklistGenes}}{}
This will \code{check for blacklisted genes}
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{gene1}} (\emph{\texttt{str}}) -- str for the name of gene at breakpoint 1

\item {} 
\textbf{\texttt{gene2}} (\emph{\texttt{str}}) -- str for the name of gene at breakpoint 2

\item {} 
\textbf{\texttt{blacklistGenes}} (\emph{\texttt{list}}) -- list containing blacklisted genes

\item {} 
\textbf{\texttt{genesToKeepFile}} (\emph{\texttt{str}}) -- str for the txt file containing genes to keep

\end{itemize}

\item[{Returns}] \leavevmode
A boolean tag indicating True or False

\item[{Return type}] \leavevmode
bool

\end{description}\end{quote}

\end{fulllineitems}

\index{checkEventInIntronFlag() (in module iCallSV.filterAnnotatedSV)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.filterAnnotatedSV.checkEventInIntronFlag}\pysiglinewithargsret{\code{iCallSV.filterAnnotatedSV.}\bfcode{checkEventInIntronFlag}}{\emph{gene1}, \emph{gene2}, \emph{site1}, \emph{site2}}{}
This will \code{Check if the event is in the intron only and not affecting
splicing}
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{gene1}} (\emph{\texttt{str}}) -- str for the name of gene at breakpoint 1

\item {} 
\textbf{\texttt{gene2}} (\emph{\texttt{str}}) -- str for the name of gene at breakpoint 2

\item {} 
\textbf{\texttt{site1}} (\emph{\texttt{str}}) -- str for the description of site in breakpoint 1

\item {} 
\textbf{\texttt{site2}} (\emph{\texttt{str}}) -- str for the description of site in breakpoint 2

\end{itemize}

\item[{Returns}] \leavevmode
A boolean tag indicating True or False

\item[{Return type}] \leavevmode
bool

\end{description}\end{quote}

\end{fulllineitems}

\index{checkGeneListToKeep() (in module iCallSV.filterAnnotatedSV)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.filterAnnotatedSV.checkGeneListToKeep}\pysiglinewithargsret{\code{iCallSV.filterAnnotatedSV.}\bfcode{checkGeneListToKeep}}{\emph{gene1}, \emph{gene2}, \emph{keepGenes}}{}
\end{fulllineitems}

\index{run() (in module iCallSV.filterAnnotatedSV)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.filterAnnotatedSV.run}\pysiglinewithargsret{\code{iCallSV.filterAnnotatedSV.}\bfcode{run}}{\emph{inputTxt}, \emph{outputDir}, \emph{outPrefix}, \emph{blacklistGenesFile}, \emph{genesToKeepFile}, \emph{verbose}}{}
This will \code{filter sv calls} from the final merged file.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{inputTxt}} (\emph{\texttt{str}}) -- str for the txt file to be filtered

\item {} 
\textbf{\texttt{outputDir}} (\emph{\texttt{str}}) -- str for the output directory

\item {} 
\textbf{\texttt{outputPrefix}} (\emph{\texttt{str}}) -- str prefix for the output File

\item {} 
\textbf{\texttt{blacklistGenesFile}} (\emph{\texttt{str}}) -- str for the txt file containing blacklisted genes

\item {} 
\textbf{\texttt{genesToKeepFile}} (\emph{\texttt{str}}) -- str for the txt file containing genes to keep

\item {} 
\textbf{\texttt{verbose}} (\emph{\texttt{bool}}) -- a boolean

\end{itemize}

\item[{Returns}] \leavevmode
A str name of final sv file

\item[{Return type}] \leavevmode
str

\end{description}\end{quote}

\end{fulllineitems}



\section{iCallSV.iCallSV module}
\label{iCallSV:icallsv-icallsv-module}\label{iCallSV:module-iCallSV.iCallSV}\index{iCallSV.iCallSV (module)}

\subsection{iCallSV}
\label{iCallSV:icallsv}\begin{quote}\begin{description}
\item[{Description}] \leavevmode
iCallSV is a wrapper to the iCallSV package which facilitates calling structural variants from Next Generation Sequencing methods such as Illumina

\item[{author}] \leavevmode
Ronak H Shah

\item[{copyright}] \leavevmode\begin{enumerate}
\setcounter{enumi}{2}
\item {} 
2015-2016 by Ronak H Shah for Memorial Sloan Kettering Cancer Center. All rights reserved.

\end{enumerate}

\item[{license}] \leavevmode
Apache License 2.0

\item[{contact}] \leavevmode
\href{mailto:rons.shah@gmail.com}{rons.shah@gmail.com}

\end{description}\end{quote}


\section{iCallSV.iCallSV\_dmp\_wrapper module}
\label{iCallSV:icallsv-icallsv-dmp-wrapper-module}\label{iCallSV:module-iCallSV.iCallSV_dmp_wrapper}\index{iCallSV.iCallSV\_dmp\_wrapper (module)}

\subsection{iCallSV\_dmp\_wrapper}
\label{iCallSV:icallsv-dmp-wrapper}\begin{quote}\begin{description}
\item[{Description}] \leavevmode
iCallSV is a wrapper to run the iCallSV package on MSKCC data

\item[{author}] \leavevmode
Ronak H Shah

\item[{copyright}] \leavevmode\begin{enumerate}
\setcounter{enumi}{2}
\item {} 
2015-2016 by Ronak H Shah for Memorial Sloan Kettering Cancer Center. All rights reserved.

\end{enumerate}

\item[{license}] \leavevmode
Apache License 2.0

\item[{contact}] \leavevmode
\href{mailto:rons.shah@gmail.com}{rons.shah@gmail.com}

\item[{deffield    updated}] \leavevmode
Updated

\end{description}\end{quote}
\index{RunJob() (in module iCallSV.iCallSV\_dmp\_wrapper)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.iCallSV_dmp_wrapper.RunJob}\pysiglinewithargsret{\code{iCallSV.iCallSV\_dmp\_wrapper.}\bfcode{RunJob}}{\emph{cmd}}{}
Given a command run the job.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\textbf{\texttt{cmd}} (\emph{\texttt{str}}) -- str of command to be run on the local machine

\item[{Returns}] \leavevmode
None

\item[{Return type}] \leavevmode
None

\end{description}\end{quote}

\end{fulllineitems}

\index{RunPerPool() (in module iCallSV.iCallSV\_dmp\_wrapper)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.iCallSV_dmp_wrapper.RunPerPool}\pysiglinewithargsret{\code{iCallSV.iCallSV\_dmp\_wrapper.}\bfcode{RunPerPool}}{\emph{titleFile}, \emph{outdir}, \emph{HSmetricsFileList}, \emph{bamFileList}, \emph{args}}{}
This will run the pool to be analyzed.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{titleFile}} (\emph{\texttt{str}}) -- str of meta information file

\item {} 
\textbf{\texttt{outdir}} (\emph{\texttt{str}}) -- str of output directory

\item {} 
\textbf{\texttt{HSmetricsFileList}} (\emph{\texttt{list}}) -- list of picard hsmetrics files

\item {} 
\textbf{\texttt{bamFileList}} (\emph{\texttt{list}}) -- list of bam files

\item {} 
\textbf{\texttt{args}} (\emph{\texttt{Namespace}}) -- Namespace of args to get other variables

\end{itemize}

\item[{Returns}] \leavevmode
None

\item[{Return type}] \leavevmode
None

\end{description}\end{quote}

\end{fulllineitems}

\index{SelectNormal() (in module iCallSV.iCallSV\_dmp\_wrapper)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.iCallSV_dmp_wrapper.SelectNormal}\pysiglinewithargsret{\code{iCallSV.iCallSV\_dmp\_wrapper.}\bfcode{SelectNormal}}{\emph{normal}, \emph{poolnormal}}{}
Select the best possible normal.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{normal}} (\emph{\texttt{str}}) -- str of match normal

\item {} 
\textbf{\texttt{poolnormal}} (\emph{\texttt{str}}) -- str of pool normal

\end{itemize}

\item[{Returns}] \leavevmode
str with decision whether to run matched or unmatched

\item[{Return type}] \leavevmode
str

\end{description}\end{quote}

\end{fulllineitems}

\index{SetupRun() (in module iCallSV.iCallSV\_dmp\_wrapper)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.iCallSV_dmp_wrapper.SetupRun}\pysiglinewithargsret{\code{iCallSV.iCallSV\_dmp\_wrapper.}\bfcode{SetupRun}}{\emph{poolName}, \emph{args}}{}
This will setup the run to be analyzed.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{poolName}} (\emph{\texttt{str}}) -- str of pool to be analyzed

\item {} 
\textbf{\texttt{args}} (\emph{\texttt{Namespace}}) -- Namespace of args to get other variables

\end{itemize}

\item[{Returns}] \leavevmode
Multiple objects

\item[{Return type}] \leavevmode
list

\end{description}\end{quote}

\end{fulllineitems}

\index{getSubDirs() (in module iCallSV.iCallSV\_dmp\_wrapper)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.iCallSV_dmp_wrapper.getSubDirs}\pysiglinewithargsret{\code{iCallSV.iCallSV\_dmp\_wrapper.}\bfcode{getSubDirs}}{\emph{dirLocation}}{}
Get all sub directories.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\textbf{\texttt{dirLocation}} (\emph{\texttt{str}}) -- str of directory location

\item[{Returns}] \leavevmode
list of all sub directories

\item[{Return type}] \leavevmode
list

\end{description}\end{quote}

\end{fulllineitems}

\index{main() (in module iCallSV.iCallSV\_dmp\_wrapper)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.iCallSV_dmp_wrapper.main}\pysiglinewithargsret{\code{iCallSV.iCallSV\_dmp\_wrapper.}\bfcode{main}}{}{}
\end{fulllineitems}

\index{processor() (in module iCallSV.iCallSV\_dmp\_wrapper)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.iCallSV_dmp_wrapper.processor}\pysiglinewithargsret{\code{iCallSV.iCallSV\_dmp\_wrapper.}\bfcode{processor}}{\emph{i}, \emph{jobqueue}}{}
Operate on a jobqueue.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{i}} (\emph{\texttt{int}}) -- count of the job

\item {} 
\textbf{\texttt{jobqueue}} (\emph{\texttt{Namespace}}) -- Namespace for jobqueue

\end{itemize}

\item[{Returns}] \leavevmode
None

\item[{Return type}] \leavevmode
None

\end{description}\end{quote}

\end{fulllineitems}



\section{iCallSV.launchThreads module}
\label{iCallSV:icallsv-launchthreads-module}\label{iCallSV:module-iCallSV.launchThreads}\index{iCallSV.launchThreads (module)}
Created on December 21, 2015
Description: This module will be launching functions as threads
@author: Ronak H Shah
\index{myThread (class in iCallSV.launchThreads)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.launchThreads.myThread}\pysiglinewithargsret{\strong{class }\code{iCallSV.launchThreads.}\bfcode{myThread}}{\emph{threadID}, \emph{name}, \emph{counter}}{}
Bases: \code{threading.Thread}
\index{run() (iCallSV.launchThreads.myThread method)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.launchThreads.myThread.run}\pysiglinewithargsret{\bfcode{run}}{}{}
\end{fulllineitems}


\end{fulllineitems}

\index{print\_time() (in module iCallSV.launchThreads)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.launchThreads.print_time}\pysiglinewithargsret{\code{iCallSV.launchThreads.}\bfcode{print\_time}}{\emph{threadName}, \emph{delay}, \emph{counter}}{}
\end{fulllineitems}



\section{iCallSV.launch\_FilterDellyCalls module}
\label{iCallSV:module-iCallSV.launch_FilterDellyCalls}\label{iCallSV:icallsv-launch-filterdellycalls-module}\index{iCallSV.launch\_FilterDellyCalls (module)}

\subsection{launch\_FilterDellyCalls}
\label{iCallSV:launch-filterdellycalls}\begin{quote}\begin{description}
\item[{Description}] \leavevmode
This module will filter delly results and create filtered delly vcf files

\end{description}\end{quote}
\index{launch\_filterdellycalls\_for\_different\_analysis\_type() (in module iCallSV.launch\_FilterDellyCalls)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.launch_FilterDellyCalls.launch_filterdellycalls_for_different_analysis_type}\pysiglinewithargsret{\code{iCallSV.launch\_FilterDellyCalls.}\bfcode{launch\_filterdellycalls\_for\_different\_analysis\_type}}{\emph{args}, \emph{config}, \emph{sampleOutdirForDelly}, \emph{del\_vcf}, \emph{dup\_vcf}, \emph{inv\_vcf}, \emph{tra\_vcf}}{}
This will launch the filtering of delly calls in parallel.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{args}} (\emph{\texttt{Namespace}}) -- Namespace of args to get other variables

\item {} 
\textbf{\texttt{config}} (\emph{\texttt{Namespace}}) -- configuration file passed to iCallSV

\item {} 
\textbf{\texttt{sampleOutdirForDelly}} (\emph{\texttt{str}}) -- Output directory for delly vcf files.

\item {} 
\textbf{\texttt{del\_vcf}} (\emph{\texttt{str}}) -- Path to deletion based vcf file

\item {} 
\textbf{\texttt{dup\_vcf}} (\emph{\texttt{str}}) -- Path to duplication based vcf file

\item {} 
\textbf{\texttt{inv\_vcf}} (\emph{\texttt{str}}) -- Path to inversion based vcf file

\item {} 
\textbf{\texttt{tra\_vcf}} (\emph{\texttt{str}}) -- Path to translocation based vcf file

\end{itemize}

\item[{Returns}] \leavevmode
Multiple objects

\item[{Return type}] \leavevmode
list

\end{description}\end{quote}

\end{fulllineitems}



\section{iCallSV.launch\_Run\_Delly module}
\label{iCallSV:module-iCallSV.launch_Run_Delly}\label{iCallSV:icallsv-launch-run-delly-module}\index{iCallSV.launch\_Run\_Delly (module)}

\subsection{launch\_Run\_Delly}
\label{iCallSV:launch-run-delly}\begin{quote}\begin{description}
\item[{Description}] \leavevmode
This module will be launching delly using Run\_Delly

\end{description}\end{quote}
\index{launch\_delly\_for\_different\_analysis\_type() (in module iCallSV.launch\_Run\_Delly)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.launch_Run_Delly.launch_delly_for_different_analysis_type}\pysiglinewithargsret{\code{iCallSV.launch\_Run\_Delly.}\bfcode{launch\_delly\_for\_different\_analysis\_type}}{\emph{args}, \emph{config}, \emph{sampleOutdirForDelly}}{}
This will launch delly calls in parallel.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{args}} (\emph{\texttt{Namespace}}) -- Namespace of args to get other variables

\item {} 
\textbf{\texttt{config}} (\emph{\texttt{Namespace}}) -- configuration file passed to iCallSV

\item {} 
\textbf{\texttt{sampleOutdirForDelly}} (\emph{\texttt{str}}) -- Output directory for delly vcf files.

\end{itemize}

\item[{Returns}] \leavevmode
Multiple objects

\item[{Return type}] \leavevmode
list

\end{description}\end{quote}

\end{fulllineitems}



\section{iCallSV.make\_analysis\_dir module}
\label{iCallSV:module-iCallSV.make_analysis_dir}\label{iCallSV:icallsv-make-analysis-dir-module}\index{iCallSV.make\_analysis\_dir (module)}

\subsection{make\_analysis\_dir}
\label{iCallSV:make-analysis-dir}\begin{quote}\begin{description}
\item[{Description}] \leavevmode
This module will make directory structure for running analysis

\end{description}\end{quote}
\index{makeOutputDir() (in module iCallSV.make\_analysis\_dir)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.make_analysis_dir.makeOutputDir}\pysiglinewithargsret{\code{iCallSV.make\_analysis\_dir.}\bfcode{makeOutputDir}}{\emph{args}, \emph{tool}}{}
This will make the output directory tree.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\textbf{\texttt{args}} (\emph{\texttt{Namespace}}) -- Namespace of args to get other variables

\item[{Returns}] \leavevmode
Multiple objects

\item[{Return type}] \leavevmode
list

\end{description}\end{quote}

\end{fulllineitems}



\section{iCallSV.makebamindex module}
\label{iCallSV:module-iCallSV.makebamindex}\label{iCallSV:icallsv-makebamindex-module}\index{iCallSV.makebamindex (module)}

\subsection{makebamindex}
\label{iCallSV:makebamindex}\begin{quote}\begin{description}
\item[{Description}] \leavevmode
Use PySAM to make bam index

\end{description}\end{quote}
\index{MakeIndex() (in module iCallSV.makebamindex)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.makebamindex.MakeIndex}\pysiglinewithargsret{\code{iCallSV.makebamindex.}\bfcode{MakeIndex}}{\emph{bamFile}}{}
This will make bam index if not there for a bam file using pysam.
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode
\textbf{\texttt{bamFile}} (\emph{\texttt{str}}) -- Path to bam file

\item[{Returns}] \leavevmode
None

\item[{Return type}] \leavevmode
None

\end{description}\end{quote}

\end{fulllineitems}



\section{iCallSV.mergeFinalFiles module}
\label{iCallSV:module-iCallSV.mergeFinalFiles}\label{iCallSV:icallsv-mergefinalfiles-module}\index{iCallSV.mergeFinalFiles (module)}

\subsection{mergeFinalFiles}
\label{iCallSV:mergefinalfiles}\begin{quote}\begin{description}
\item[{Description}] \leavevmode
Merge VCF, iAnnotateSV tab and targetSeqView tab file into a single tab-delimited file

\end{description}\end{quote}
\index{run() (in module iCallSV.mergeFinalFiles)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.mergeFinalFiles.run}\pysiglinewithargsret{\code{iCallSV.mergeFinalFiles.}\bfcode{run}}{\emph{aId}, \emph{bId}, \emph{vcfFile}, \emph{annoTab}, \emph{confTab}, \emph{outDir}, \emph{outputPrefix}, \emph{verbose}}{}
This will Merge VCF, iAnnotateSV tab and targetSeqView tab file into a single tab-delimited file
\begin{quote}\begin{description}
\item[{Parameters}] \leavevmode\begin{itemize}
\item {} 
\textbf{\texttt{aId}} (\emph{\texttt{str}}) -- Sample ID for case that has the structural abberations

\item {} 
\textbf{\texttt{bId}} (\emph{\texttt{str}}) -- Sample ID for control

\item {} 
\textbf{\texttt{vcfFile}} (\emph{\texttt{str}}) -- Delly filtered and merged VCF file

\item {} 
\textbf{\texttt{annoTab}} (\emph{\texttt{str}}) -- iAnnotateSV tab-delimited file with annotations

\item {} 
\textbf{\texttt{confTab}} (\emph{\texttt{str}}) -- targetSeqView tab-delimited file with probability score

\item {} 
\textbf{\texttt{outputDir}} (\emph{\texttt{str}}) -- Directory to write the output file

\item {} 
\textbf{\texttt{outputPrefix}} (\emph{\texttt{str}}) -- Output File Prefix

\end{itemize}

\item[{Returns}] \leavevmode
str of the tab-delimited file

\item[{Return type}] \leavevmode
str

\end{description}\end{quote}

\end{fulllineitems}



\section{iCallSV.sortbamByCoordinate module}
\label{iCallSV:module-iCallSV.sortbamByCoordinate}\label{iCallSV:icallsv-sortbambycoordinate-module}\index{iCallSV.sortbamByCoordinate (module)}

\subsection{sortbamByCoordinate}
\label{iCallSV:sortbambycoordinate}\begin{quote}\begin{description}
\item[{Description}] \leavevmode
This module will sort bam file by coordinate

\end{description}\end{quote}
\index{sortBam() (in module iCallSV.sortbamByCoordinate)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.sortbamByCoordinate.sortBam}\pysiglinewithargsret{\code{iCallSV.sortbamByCoordinate.}\bfcode{sortBam}}{\emph{inputBam}, \emph{outputBamName}, \emph{outputDir}}{}
\end{fulllineitems}



\section{iCallSV.sortbamByReadName module}
\label{iCallSV:module-iCallSV.sortbamByReadName}\label{iCallSV:icallsv-sortbambyreadname-module}\index{iCallSV.sortbamByReadName (module)}

\subsection{sortbamByReadName}
\label{iCallSV:sortbambyreadname}\begin{quote}\begin{description}
\item[{Description}] \leavevmode
This module will sort bam file by name

\end{description}\end{quote}
\index{sortBam() (in module iCallSV.sortbamByReadName)}

\begin{fulllineitems}
\phantomsection\label{iCallSV:iCallSV.sortbamByReadName.sortBam}\pysiglinewithargsret{\code{iCallSV.sortbamByReadName.}\bfcode{sortBam}}{\emph{inputBam}, \emph{outputBamName}, \emph{outputDir}}{}
\end{fulllineitems}



\section{Module contents}
\label{iCallSV:module-contents}\label{iCallSV:module-iCallSV}\index{iCallSV (module)}

\chapter{Indices and tables}
\label{index:indices-and-tables}\begin{itemize}
\item {} 
\DUspan{xref,std,std-ref}{genindex}

\item {} 
\DUspan{xref,std,std-ref}{modindex}

\item {} 
\DUspan{xref,std,std-ref}{search}

\end{itemize}


\renewcommand{\indexname}{Python Module Index}
\begin{theindex}
\def\bigletter#1{{\Large\sffamily#1}\nopagebreak\vspace{1mm}}
\bigletter{i}
\item {\texttt{iCallSV}}, \pageref{iCallSV:module-iCallSV}
\item {\texttt{iCallSV.checkBlackList}}, \pageref{iCallSV:module-iCallSV.checkBlackList}
\item {\texttt{iCallSV.checkHotSpotList}}, \pageref{iCallSV:module-iCallSV.checkHotSpotList}
\item {\texttt{iCallSV.checkparameters}}, \pageref{iCallSV:module-iCallSV.checkparameters}
\item {\texttt{iCallSV.combineVCF}}, \pageref{iCallSV:module-iCallSV.combineVCF}
\item {\texttt{iCallSV.dellyVcf2Tab}}, \pageref{iCallSV:module-iCallSV.dellyVcf2Tab}
\item {\texttt{iCallSV.dellyVcf2targetSeqView}}, \pageref{iCallSV:module-iCallSV.dellyVcf2targetSeqView}
\item {\texttt{iCallSV.filterAnnotatedSV}}, \pageref{iCallSV:module-iCallSV.filterAnnotatedSV}
\item {\texttt{iCallSV.FilterDellyCalls}}, \pageref{iCallSV:module-iCallSV.FilterDellyCalls}
\item {\texttt{iCallSV.iCallSV}}, \pageref{iCallSV:module-iCallSV.iCallSV}
\item {\texttt{iCallSV.iCallSV\_dmp\_wrapper}}, \pageref{iCallSV:module-iCallSV.iCallSV_dmp_wrapper}
\item {\texttt{iCallSV.launch\_FilterDellyCalls}}, \pageref{iCallSV:module-iCallSV.launch_FilterDellyCalls}
\item {\texttt{iCallSV.launch\_Run\_Delly}}, \pageref{iCallSV:module-iCallSV.launch_Run_Delly}
\item {\texttt{iCallSV.launchThreads}}, \pageref{iCallSV:module-iCallSV.launchThreads}
\item {\texttt{iCallSV.make\_analysis\_dir}}, \pageref{iCallSV:module-iCallSV.make_analysis_dir}
\item {\texttt{iCallSV.makebamindex}}, \pageref{iCallSV:module-iCallSV.makebamindex}
\item {\texttt{iCallSV.mergeFinalFiles}}, \pageref{iCallSV:module-iCallSV.mergeFinalFiles}
\item {\texttt{iCallSV.Run\_Delly}}, \pageref{iCallSV:module-iCallSV.Run_Delly}
\item {\texttt{iCallSV.Run\_iAnnotateSV}}, \pageref{iCallSV:module-iCallSV.Run_iAnnotateSV}
\item {\texttt{iCallSV.Run\_samblaster}}, \pageref{iCallSV:module-iCallSV.Run_samblaster}
\item {\texttt{iCallSV.Run\_targetSeqView}}, \pageref{iCallSV:module-iCallSV.Run_targetSeqView}
\item {\texttt{iCallSV.sortbamByCoordinate}}, \pageref{iCallSV:module-iCallSV.sortbamByCoordinate}
\item {\texttt{iCallSV.sortbamByReadName}}, \pageref{iCallSV:module-iCallSV.sortbamByReadName}
\end{theindex}

\renewcommand{\indexname}{Index}
\printindex
\end{document}
